exporter/opentelemetry-exporter-prometheus-remote-write/src/opentelemetry/exporter/prometheus_remote_write/__init__.py

```diff
@@ -142,50 +153,144 @@ def export(
     def shutdown(self) -> None:
         raise NotImplementedError()
 
-    def convert_to_timeseries(
+    def _convert_to_timeseries(
         self, export_records: Sequence[ExportRecord]
     ) -> Sequence[TimeSeries]:
-        raise NotImplementedError()
-
-    def convert_from_sum(self, sum_record: ExportRecord) -> TimeSeries:
-        raise NotImplementedError()
-
-    def convert_from_min_max_sum_count(
+        converter_map = {
+            MinMaxSumCountAggregator: self._convert_from_min_max_sum_count,
+            SumAggregator: self._convert_from_sum,
+            HistogramAggregator: self._convert_from_histogram,
+            LastValueAggregator: self._convert_from_last_value,
+            ValueObserverAggregator: self._convert_from_value_observer,
+        }
+        timeseries = []
+        for export_record in export_records:
+            aggregator_type = type(export_record.aggregator)
+            converter = converter_map.get(aggregator_type)
+            if converter is None:
+                raise ValueError(
+                    str(aggregator_type) + " conversion is not supported"
+                )
+            timeseries.extend(converter(export_record))
+        return timeseries
+
+    def _convert_from_sum(self, sum_record: ExportRecord) -> TimeSeries:
+        return [
+            self._create_timeseries(
+                sum_record,
+                sum_record.instrument.name,
+                sum_record.aggregator.checkpoint,
+            )
+        ]
+
+    def _convert_from_min_max_sum_count(
         self, min_max_sum_count_record: ExportRecord
     ) -> TimeSeries:
-        raise NotImplementedError()
-
-    def convert_from_histogram(
+        timeseries = []
+        for agg_type in ["min", "max", "sum", "count"]:
+            name = min_max_sum_count_record.instrument.name + "_" + agg_type
+            value = getattr(
+                min_max_sum_count_record.aggregator.checkpoint, agg_type
+            )
+            timeseries.append(
```

Comment:
Prometheus accepts a separate timeseries for each of these values?

Reply:
Prometheus doesn't, but backends like Cortex do (exported from Prometheus). Cortex only accepts the TimeSeries format which leaves no other way to store aggregations of different types than creating multiple TimeSeries with different labels.

Reply:
What about other backends that support Prometheus data format? Seems like this would be a CortexPrometheusRemoteWriteExporter in this case then.

Reply:
Sorry, what I meant was that although Prometheus does not receive data in TimeSeries format when using Prometheus instance and scraping metric_data, it's remote write api can only send data in TimeSeries format to an integrated backend. Therefore, any backend (influxDB, Thanos, Cortex) will need to receive the more complex aggregations split into separate TimeSeries with clear labelling. Cortex was just an example.